synapse/handlers/oidc.py

```diff
@@ -1112,6 +1456,7 @@ def _get_secret(self) -> bytes:
         logger.info(
             "Generating new JWT for %s: %s %s", self._oauth_issuer, header, payload
         )
+        jwt = JsonWebToken(header["alg"])
```

Comment:
What's this change for? Is it ensuring `JsonWebToken` doesn't explode?

Reply:
we were importing `authlib.jose.jwt` before and it's not the case anymore. I think I did this because it was not available in all versions of authlib?